packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -185,27 +185,20 @@ class FlutterPlugin implements Plugin<Project> {
 
         File pluginsFile = new File(project.projectDir.parentFile.parentFile, '.flutter-plugins')
         Properties plugins = readPropertiesIfExist(pluginsFile)
-
-        plugins.each { name, _ ->
-            def pluginProject = project.rootProject.findProject(":$name")
-            if (pluginProject != null) {
-                project.dependencies {
-                    if (project.getConfigurations().findByName("implementation")) {
-                        implementation pluginProject
-                    } else {
-                        compile pluginProject
+        // Plugins right now only have a notion of debug or release.
```

Comment:
Is there anything preventing us from generalizing this beyond debug and release?  I think we already have developers who are struggling with flavor support, so if we can do a little extra to ensure that this supports arbitrary modes, that would probably be ideal.

Reply:
This is an existing restriction.  I'm fine with looking at ways to tackle this, but probably in a separate PR.